src/main/java/cz/it4i/parallel/ImageJServerWorker.java

```diff
@@ -46,9 +46,14 @@
 	private final static Set<String> supportedImageTypes = Collections
 			.unmodifiableSet(new HashSet<>(Arrays.asList("png", "jpg")));
 
-	ImageJServerWorker(String hostName, int port) {
+	ImageJServerWorker(String hostName, int defaultPort) {
 		this.hostName = hostName;
-		this.port = port;
+		if(hostName.contains(":")) {
```

Comment:
I think that this logic should either be moved upstream to initWorkerPool().

Reply:
You, are right - it was easier to implement it in the worker.